.circleci/config.yml

```diff
@@ -1,25 +1,109 @@
-version: '2.1'
+version: 2.1
 orbs:
-  flutter: circleci/flutter@1.0
+  flutter: circleci/flutter@1.1.0
+  android: circleci/android@1.0.3
 jobs:
-  distribute:
+  test_iOS:
     macos:
-      xcode: 12.5.1
+      xcode: 13.3.1
     steps:
+      - checkout
       - flutter/install_sdk_and_pub:
-          version: 2.2.3
-      - flutter/install_ios_pod
-      - flutter/install_ios_gem
+          flutter_version: 2.10.5
+          app-dir: example
+      - flutter/install_ios_pod:
+          app-dir: example
       - run:
-          command: bundle exec fastlane distribute
-          working_directory: ios
+          command: flutter test
+          working_directory: example
+  publish_package:
+    macos:
+      xcode: 13.3.1
+    steps:
+      - checkout
+      - flutter/install_sdk_and_pub:
+          flutter_version: 2.10.5
+          app-dir: example
+      - run:
+          name: Install dart
+          command: |
+            brew tap dart-lang/dart
+            brew install dart
+      - run:
+          name: setup pub login
+          command: sh .circleci/pub_login.sh
+      - run:
+          name: publish pub
+          command: yes | dart pub publish
+  test_Android:
+    executor:
+      name: android/android-machine
+    steps:
+      - checkout
+      - flutter/install_sdk_and_pub:
+          flutter_version: 2.10.5
+          app-dir: example
+      - flutter/install_android_gradle:
+          app-dir: example
+      - run:
+          name: build apk
+          command: flutter build apk
+          working_directory: example
+      - run:
+          name: Run tests
+          command: flutter test
+          working_directory: example
 workflows:
-  test:
+  version: 2
+  build_test:
+    jobs:
+      - flutter/lint:
+          version: 2.10.5
+          filters:
+            branches:
+              ignore:
+                - /release-(.*)/
+      - test_iOS:
+          filters:
+            branches:
+              ignore:
+                - /release-(.*)/
+      - test_Android:
+          filters:
+            branches:
+              ignore:
+                - /release-(.*)/
+  publish_plugin:
     jobs:
-      - flutter/unit_test:
-          version: 2.2.3
       - flutter/lint:
-          version: 2.2.3
-  # distribute:
-  #   jobs:
-  #     - distribute
\ No newline at end of file
+          version: 2.10.5
+          filters:
+            branches:
+              only:
+                - /release-(.*)/
+      - test_iOS:
```

Comment:
A question, what if the lint job fails? 
Shouldn't test_iOS and test_Android depend on it? This way publish job will also not run.
Same in build_test workflow as well.

Reply:
Yes but this approach is faster as lint will be triggered concurrently.
If there are any build failures, we cannot merge the PR so it matters if we want to do it quickly or save some CircleCI credit.
I could change it

Reply:
yes, I feel better to change, we should not build if lint has failed.

Reply:
done